front/src/applications/stdcmV2/hooks/useStdmcEnv.tsx

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useState } from 'react';
+
+import { useDispatch } from 'react-redux';
+
+import { osrdEditoastApi } from 'common/api/osrdEditoastApi';
+import { useOsrdConfActions } from 'common/osrdContext';
+
+export const NO_CONFIG_FOUND_MSG = 'No configuration found';
+
+export default function useStdcmEnvironment() {
+  const dispatch = useDispatch();
+  const {
+    updateInfraID,
+    updateTimetableID,
+    updateElectricalProfileSetID,
+    updateWorkScheduleGroupID,
+    updateSearchDatetimeWindow,
+  } = useOsrdConfActions();
+  const [getStdcmSearchEnvironment] =
+    osrdEditoastApi.endpoints.getStdcmSearchEnvironment.useLazyQuery();
+
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState<null | Error>(null);
+
+  const loadStdcmEnvironment = useCallback(async () => {
```

Comment:
It's only a question but do we need to create this async function or do you think we could use a simple useQuery (with the state loading and error of the request) ?

I'm fine with this solution too

Reply:
We can use the `useQuery` and check the error, the loading and do the dispatch.
But in this case all the code of this hook should be included in the `StdcmViewv2`  component.

I did it to centralized the logic in one file.

Reply:
You can define the function inside the useEffect directly